Pre-Trainee Task/Controllers/FeedbackController.cs

```
@@ -152,13 +134,18 @@ public ActionResult<FeedbackReadDto> UpdateFeedback(Guid id,
     /// <response code="404">Feedback with the given ID was not found</response>
     [Authorize(Roles = "Admin")]
     [HttpDelete("{id}")]
-    public ActionResult DeleteFeedback(Guid id)
+    public async Task<ActionResult> DeleteFeedbackAsync(Guid id)
     {
-        var feedback = _feedbackService.GetById(id);
-        if (_feedbackService.Delete(id)) return NoContent();
+        var feedback = await _feedbackService.GetFeedbackByIdAsync(id);
+        var deleted = await _feedbackService.DeleteAsync(id);
+        if (!deleted) return NotFound();
 
-        SendEmail("admin@admin.com", $"Feedback {id} deleted",
-            $"Deleted feedback content: {feedback}");
-        return NotFound();
+        SendEmailAsync("admin@admin.com", $"Feedback {id} deleted", $"Deleted feedback content: {feedback}");
+        return NoContent();
+    }
+    
+    private void SendEmailAsync(string email, string subject, string body)
+    {
+        _ = Task.Run(async () => await _emailService.SendAsync(email, subject, body));
```

Avoid using `Task.Run` for async operations in controllers. It may suppress exceptions and cause background execution issues. Just make the method itself async and await it directly